Src/BootCamp.Chapter/BalanceStats.cs

```diff
@@ -1,37 +1,236 @@
-﻿namespace BootCamp.Chapter
+﻿using System;
+using System.Globalization;
+using System.Threading;
+using static System.FormattableString;
+
+
+namespace BootCamp.Chapter
 {
     public static class BalanceStats
     {
+        
         /// <summary>
         /// Return name and balance(current) of person who had the biggest historic balance.
         /// </summary>
         public static string FindHighestBalanceEver(string[] peopleAndBalances)
         {
-            return "";
+            if (peopleAndBalances == null || peopleAndBalances.Length == 0)
+            {
+                return "N/A.";
+            }
+
+            double highestBalance = double.MinValue;
+            string highestBalanceHolder = "";
+            int holderCount = 0;
+
+            foreach (string entry in peopleAndBalances)
+            {
+                string[] userNameAndBalanceHistory = entry.Split(',');
+                string currentBalanceHolder = userNameAndBalanceHistory[0];
+
+                for (int i = 1; i < userNameAndBalanceHistory.Length; i++)
+                {
+                    double balance;
+                    try
+                    {
+                        balance = double.Parse(userNameAndBalanceHistory[i]);
+                    }
+                    catch
+                    {
+                        balance = 0;
+                    }
+                    if (balance > highestBalance)
+                    {
+                        highestBalanceHolder = currentBalanceHolder;
+                        holderCount = 1;
+                    }
+                    if (balance == highestBalance)
+                    {
+                        if (holderCount == 2)
+                        {
+                            highestBalanceHolder += ($" and {currentBalanceHolder}");
+                        }
+                        if (holderCount == 1)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Refactor this code to not nest more than 3 control flow statements.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4953253)